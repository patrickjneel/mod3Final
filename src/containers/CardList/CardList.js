import React, { Component } from 'react';
import { connect } from 'react-redux';

class CardList extends Component {

  componentDidMount() {

  }

  render() {
    return (
      <div className="card-list">
      Cards Go Here
      </div>
    )
  }
}

export default CardList;
