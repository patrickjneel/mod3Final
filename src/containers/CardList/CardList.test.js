import React from 'react';
import { shallow } from 'enzyme';
import { CardList, mapStateToProps, mapDispatchToProps} from './CardList';

describe('CardList Test', () => {
  
  it('should be defined', () => {
    const mockFunc = jest.fn();
    const data = {name: 'stark', words: 'winter is coming'}
    const renderedCardList = shallow(<CardList data={mockFunc} />, {disableLifecycleMethods: true})
    console.log(renderedCardList)
    expect(renderedCardList).toBeDefined();
  })

  it('should match snap shot', () => {
    const mockFunc = jest.fn();
    const renderedCardList = shallow(<CardList data={mockFunc}/>, {disableLifecycleMethods: true});
    expect(renderedCardList).toMatchSnapshot();
  });

});

describe('mapStateToProps test', () => {
  it('should receive props from the store', () => {
    const mockState = {
      data: [{name: 'Stark', words: 'Winter is Coming', weapons: 'Ice'}]
    }
    const expected = mapStateToProps(mockState)
    expect(expected.data).toEqual(mockState.data)
  })
})

describe('mapDispatchToProps test', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

  it('should call dispatch when addHouses is called', () => {

    result.addHouses();
    expect(mockDispatch).toHaveBeenCalled();
  })
})
