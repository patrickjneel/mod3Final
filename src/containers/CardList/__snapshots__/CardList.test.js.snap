// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Test should match snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    data={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "loading...",
    },
    "ref": null,
    "rendered": "loading...",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "loading...",
      },
      "ref": null,
      "rendered": "loading...",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
